game/src/AI/States/Destroyed.gd

```diff
@@ -0,0 +1,30 @@
+extends State
+"""
+State to make enemy fade to nothing, then get released by the scene tree.
+"""
+
+
+export var hurt_color: Color
+
+
+func enter(msg: Dictionary = {}) -> void:
+	owner.hook_target.set_color(Color(0, 0, 0, 0))
+	owner.body.set_color_fill(hurt_color)
+	
+	yield(do_color(), "completed")
+	
+	owner.queue_free()
+
+
+func do_color() -> void:
```

Comment:
Use an animation player for things like these, you're making your life harder than it needs to be, and in general the anim player will likely have better perfs than any GDScript code